### PR TITLE
Remove cyclical imports

### PR DESCRIPTION
**Pull request by christopherdcunha**
### Motivation

This gets rid of the cyclical imports described in https://github.com/callstack/react-native-paper/issues/639 and https://github.com/callstack/react-native-paper/issues/930

### Test plan

Existing tests should pass. This PR should not add new functionality.

### CODE REVIEW

**Comment by ferrannp** [comment]
@christopherdcunha what's the status of this PR?

File: src/components/Appbar/index.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  static Content = AppbarContent;
+  // @component ./AppbarAction.js
+  static Action = AppbarAction;
+  // @component ./AppbarBackAction.js
+  static BackAction = AppbarBackAction;
+  // @component ./AppbarHeader.js
+  static Header = AppbarHeader;
+
+  isAppbarContent = element => element.type === AppbarContent
+  isValidElement = element => (
```
Comment (ferrannp) [inline src/components/Appbar/index.js]: Can you define them as function on the top instead of adding them as class properties?

**Comment by christopherdcunha** [comment]
@ferrannp Thanks for having a look at this. I created this PR to propose a way to get rid of the cyclical imports described in #639 and #930. If this approach looks sensible to the core contributors, then I'll go ahead an implement the full thing.

I'll try to explain the concept in words (though I think the code does a better job).

Currently you have a bunch of components that depend on each other. Let's say you have A and B, where A is a function of B, and B is a function of A. My proposal is remove the dependency from one of these (or maybe both) and create a new component (C) that is a function of A and B.

**Comment by BransonGitomeh** [comment]
Hello guys, anyone actively working on this?

**Comment by christopherdcunha** [comment]
Hi @BransonGitomeh, yes I plan on submitting a more complete PR for this soon.

File: src/components/Appbar/index.js (hunk 1 above)
Comment (christopherdcunha) [reply src/components/Appbar/index.js]: Sorry for the delayed reply @ferrannp. I ended up changing the implementation quite a bit when I realized certain things about the imports (mainly that only a few imports result in circular imports). I don't need these functions anymore.

**Comment by christopherdcunha** [comment]
@satya164, in case this isn't on your radar, I want to let you know that this PR is ready to be reviewed.

**Comment by raajnadar** [comment]
@christopherdcunha can you fix the merge conflicts?

**Comment by jaysbytes** [comment]
What is the status of the PR? It fixes the issue in only one component and @christopherdcunha you said you are working on more complete PR.

**Comment by christopherdcunha** [comment]
@jaysbytes Yes, I did indeed commit to rebase and adapt the code to work post-TypeScript migration. Life happened and I haven't worked on it, but I should have time to do this soon (probably by the end of the month).

**Comment by jaysbytes** [comment]
@christopherdcunha ok, will keep it open for a while.

**Comment by christopherdcunha** [comment]
@BransonGitomeh @Javi Sorry, but I'm going to close this PR as I don't have a nice solution to this at the moment. I might have time to deal with this later in the year but I can't commit to anything unfortunately 😞

**Comment by Trancever** [comment]
@christopherdcunha No worries! Thanks anyway

**Pull request closed without merging**